03week/towerOfHanoi_crap.js

```diff
@@ -0,0 +1,171 @@
+'use strict';
+
+const assert = require('assert');
+const readline = require('readline');
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+
+const stacks = {
+  a: [4, 3, 2, 1],
+  b: [],
+  c: []
+};
+
+let activeStacks = stacks;
+
+// let activeStackOne = Object.values(stacks['a']);
+// let activeStackTwo = Object.values(stacks['b']);
+// let activeStackThree = Object.values(stacks['c']);
+
+function printStacks() {
+  console.log("a: " + activeStacks.a);
+  console.log("b: " + activeStacks.b);
+  console.log("c: " + activeStacks.c);
+  //console.log(stacks)
+}
+
+// const isTheStackEmpty = (startStack) => {
+  
+//   if (activeStacks[startStack] === undefined || activeStacks[startStack].length == 0) {
+//     console.log('this array is empty');
+//     return true;
+//   } 
+// }
+let startPop;
+let endPush;
+
+function movePiece(startStack, endStack) {
+  
+  // Your code here
+  // take last number from start stack array and put at end of end stack array
+  const startPop = activeStacks[startStack].pop();
+  console.log(startPop);
+  const endPush = activeStacks[endStack].push(startPop);
+  console.log(endPush);
+
+}
+
+// const properMove = (startPop,endPush) => {
+//   if(startPop < endPush) {
+//     return true;
+//   }
+// }
+
+function isLegal(startPop,endPush,endStack) {
+  // Your code here
+  if (activeStacks[endStack] === undefined) {
+    console.log('this is triggering');
+    return true;
+    
+  } 
+
+  if (startPop > endPush) {
+    return false;
+  }
+  // if(isTheStackEmpty()) {
+  //   console.log('Please move from a non-empty stack.');
+  // } else {
+  //   properMove();
+  // }
+  // start stack cannot be empty
+  // move cannot go onto end stack if last number in the end stack array is less than itself
+
+}
+
+const checkForWin = (activeStacks) => {
+  // Your code here
+  // user wins if c: 4,3,2,1
+  //const lastStack = activeStacks[];
+  if (activeStacks === {a:[],b:[],c:[4,3,2,1]}) {
```

Comment:
this would never return true, no two objects equal each other, what's a better way to check for a win?